allensdk/brain_observatory/behavior/running_processing.py

```diff
@@ -14,40 +16,378 @@ def calc_deriv(x, time):
+
+
+def _shift(
+        arr: Iterable,
+        periods: int = 1,
+        fill_value: float = np.nan) -> np.ndarray:
+    """
+    Shift index of an iterable (array-like) by desired number of
+    periods with an optional fill value (default = NaN).
+
+    Parameters
+    ----------
+    arr: Iterable (array-like)
+        Iterable containing numeric data. If int, will be converted to
+        float in returned object.
+    Returns
+    -------
+    np.ndarray (1d)
+        Copy of input object as a 1d array, shifted.
+    """
+    if fill_value is None:
+        fill_value = np.nan
+    if isinstance(fill_value, float):
+        # Circumvent issue if int-like array with np.nan as fill
+        shifted = np.roll(arr, periods).astype(float)
+    else:
+        shifted = np.roll(arr, periods)
+    shifted[:periods] = fill_value
+    return shifted
+
+
+def deg_to_dist(angular_speed: np.ndarray) -> np.ndarray:
+    """
+    Takes the angular speed (radians/s) at each step in radians, and
+    computes the linear speed in cm/s.
+
+    Parameters
+    ----------
+    angular_speed: np.ndarray (1d)
+        1d array of angular speed in radians/s
+    Returns
+    -------
+    np.ndarray (1d)
+        Linear speed in cm/s at each time point.
+    """
     wheel_diameter = 6.5 * 2.54  # 6.5" wheel diameter
     running_radius = 0.5 * (
-        2.0 * wheel_diameter / 3.0)  # assume the animal runs at 2/3 the distance from the wheel center
-    running_speed_cm_per_sec = np.pi * speed_deg_per_s * running_radius / 180.
+        # assume the animal runs at 2/3 the distance from the wheel center
+        2.0 * wheel_diameter / 3.0)
+    running_speed_cm_per_sec = angular_speed * running_radius
     return running_speed_cm_per_sec
 
 
-def get_running_df(data, time):
-    dx_raw = data["items"]["behavior"]["encoders"][0]["dx"]
+def _identify_wraps(vsig: Iterable, *,
+                    min_threshold: float = 1.5,
+                    max_threshold: float = 3.5):
+    """
+    Identify "wraps" in the voltage signal. In practice, this is when
+    the encoder voltage signal crosses 5V and wraps to 0V, or
+    vice-versa.
+
+    Argument defaults and implementation suggestion via @dougo
+
+    Parameters
+    ----------
+    vsig: Iterable (array-like)
+        1d array-like iterable of voltage signal
+    min_threshold: float (default=1.5)
+        The min_threshold value that must be crossed to be considered
+        a possible wrapping point.
+    max_threshold: float (default=3.5)
+        The max threshold value that must be crossed to be considered
+        a possible wrapping point.
+
+    Returns
+    -------
+    Tuple
+        Tuple of ([indices of positive wraps], [indices of negative wraps])
+    """
+    # Compare against previous value
+    shifted_vsig = _shift(vsig)
+    if not isinstance(vsig, np.ndarray):
+        vsig = np.array(vsig)
+    # Suppress warnings for when comparing to nan values
+    with np.errstate(invalid='ignore'):
```

Comment:
this could be simpler:
```
In [27]: vsig = np.array([0.0, 0.1, 0.4, 1.4, 2.5, 3.6, 4.8, 5.0, 0.0, 0.1, 4.9, 0.3, 2.0, 3.0, 4.0, 4.9, 0.0])

In [28]: shifted = np.roll(vsig, 1)

In [29]: vsig
Out[29]: 
array([0. , 0.1, 0.4, 1.4, 2.5, 3.6, 4.8, 5. , 0. , 0.1, 4.9, 0.3, 2. ,
       3. , 4. , 4.9, 0. ])

In [30]: shifted
Out[30]: 
array([0. , 0. , 0.1, 0.4, 1.4, 2.5, 3.6, 4.8, 5. , 0. , 0.1, 4.9, 0.3,
       2. , 3. , 4. , 4.9])

In [31]: pos_wraps = np.argwhere((vsig < 1.5) & (shifted > 3.5))

In [32]: pos_wraps
Out[32]: 
array([[ 8],
       [11],
       [16]])

In [33]: neg_wraps = np.argwhere((vsig > 3.5) & (shifted < 1.5))

In [34]: neg_wraps
Out[34]: array([[10]])

In [35]: pos_wraps
Out[35]: 
array([[ 8],
       [11],
       [16]])

In [36]: pos_wraps.flatten()
Out[36]: array([ 8, 11, 16])
```

Reply:
I used `nonzero` because of this documentation on argwhere: `The output of argwhere is not suitable for indexing arrays. For this purpose use nonzero(a) instead.`

Reply:
It's the context manager for handling nans that I am flagging as unusual. I don't think you need this.

Reply:
Oh, I see. The context manager is because the methods were raising warnings when comparing against NaN, and Doug said that it was too noisy.

Reply:
I'm still getting the warning from your example when NaNs are included, so I'm going to leave it as-is if that's ok.

Reply:
Sure, this just seems not an elegant way to handle missing data. I think this is what the `numpy.ma` module is for. But, if you just need to move on, ok.